great_expectations/core/configuration.py

```diff
@@ -1,10 +1,26 @@
 """Contains general abstract or base classes used across configuration objects."""
 from abc import ABC
-from typing import Optional
+from typing import Mapping, Optional, cast
+
+from great_expectations.marshmallow__shade import Schema
+
+try:
+    from typing import Protocol
+except ImportError:
+    from typing_extensions import Protocol  # type: ignore[misc]
 
 from great_expectations.types import SerializableDictDot
 
 
+# TODO: get rid of protocol if we aren't using it
+class Loadable(Protocol):
+    """Object should have a `.load(x)` method that returns a dict-like `Mapping` object."""
+
+    def load(self, data) -> Mapping:
+        """Load an object and return a `dict` or dict-like `Mapping` object."""
+        ...
+
+
 class AbstractConfig(ABC, SerializableDictDot):
     """Abstract base class for Config objects. Sets the fields that must be included on a Config."""
 
```

Comment:
Note: use `id` instead of `id_` in constructor and then remove shadowed `id` builtin

1. id field??
2. look for `id_`